source/extensions/common/dynamic_forward_proxy/dns_cache_impl.h

```diff
@@ -132,7 +132,7 @@ class DnsCacheImpl : public DnsCache, Logger::Loggable<Logger::Id::forward_proxy
     void setAddresses(Network::Address::InstanceConstSharedPtr address,
                       std::vector<Network::Address::InstanceConstSharedPtr>&& list) {
       absl::WriterMutexLock lock{&resolve_lock_};
-      first_resolve_complete_ = true;
+      first_resolve_complete_ = false;
```

Comment:
With the above added line to set this flag outside this function, you can simply remove this line.

Reply:
thanks, remove this line in latest change